src/languageservice/services/validation/types.ts

```diff
@@ -0,0 +1,12 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { Diagnostic } from 'vscode-languageserver-types';
+import { SingleYAMLDocument } from '../../parser/yaml-documents';
+
+export interface AdditionalValidator {
```

Comment:
I am not sure about the "Additional" validators. What makes these validators additional? We are not really treating them any different, they can't be disabled. Should we treat every validation the same? Meaning the jsonvalidator is also wrapped as a `Validator` interface?